samples-generator/lib/createBuildingsTile.js

```diff
@@ -91,24 +95,54 @@ function createBuildingsTile(options) {
         featureTableJson.RTC_CENTER = Cartesian3.pack(batchedMesh.getCenter(), new Array(3));
     }
 
-    return createGltf({
+    var gltfOptions = {
         mesh : batchedMesh,
         useBatchIds : useBatchIds,
         relativeToCenter : relativeToCenter,
-        deprecated : deprecated1 || deprecated2
-    }).then(function(glb) {
-        var b3dm = createB3dm({
-            glb : glb,
+        deprecated : deprecated1 || deprecated2,
+        useGlb: useGlb,
+        useGltf: useGltf,
+        deferGlbConversion: useGlb || useGltf // TODO: This is ugly, but allows us to refactor this
+                                              //       without breaking other functions that potentially always expect
+                                              //       createGltf to return a promise that resolves to a glb.
+                                              //       Instead, it returns a promise that resolves to a gltf
+                                              //       hence 'glbOrGltf'. We can do further post-processing on the gltf,
+                                              //       and convert it to a GLB when necessary later.
```

Comment:
Maybe there should be a `createGltf` and `createGlb` that take the same options. `createGlb` could just call `createGltf` and do the conversion. Rather than listing all the options twice `createGlb` doc could look like https://github.com/CesiumGS/gltf-pipeline/blob/master/lib/gltfToGlb.js#L11-L17.

Reply:
they're split now